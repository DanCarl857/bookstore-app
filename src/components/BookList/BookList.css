.table .check {
    text-align: center;
}

.table .icon {
    color: #252525;
}
