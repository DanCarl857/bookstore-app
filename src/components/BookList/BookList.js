import React from 'react';
import { graphql } from 'react-apollo';
import { getAllBooksQuery } from './../../queries/queries';
import { Table, Input, Row, Col, Button } from 'reactstrap';
import './BookList.css';

const BookList = ({ data }) => {

    const displayBooks = () => {
        if (data.loading) {
            return (<div>Getting books...</div>);
        } else {
            return data.books.map(book => {
                return (
                    <tr key={book.bookId}>
                        <td className="check">
                            <Input type="checkbox" id="checkbox2" />
                        </td>
                        <td>{book.bookId}</td>
                        <td>{book.title}</td>
                        <td>{book.author}</td>
                        <td>$ {book.price}</td>
                        <td><i class="icon ion-md-create"></i></td>
                    </tr>
                )
            })
        }
    }

    return (
        <div className="box container">
            <Row>
                <Col xs="12" md="6">
                    <h3>Dashboard</h3>
                </Col>
                <Col xs="12" md="6">
                    <Button color="success" size="sm" className="float-right">Create New Book</Button>
                </Col>
            </Row>
            <Table striped>
                <thead>
                <tr>
                    <th>Check</th>
                    <th>ID</th>
                    <th>Title</th>
                    <th>Author Name</th>
                    <th>Price</th>
                    <td>Action</td>
                </tr>
                </thead>
                <tbody>
                    {displayBooks()}
                </tbody>
            </Table>
        </div>
    )
}

export default graphql(getAllBooksQuery)(BookList);