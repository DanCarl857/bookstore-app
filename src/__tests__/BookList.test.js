import React from 'react';
import { shallow } from 'enzyme';

import BookList from './../components/BookList/BookList';

describe('Book List', () => {
    it('should render correctly', () => {
        const component = shallow(<BookList />);
        expect(component).toMatchSnapshot();
    })
});