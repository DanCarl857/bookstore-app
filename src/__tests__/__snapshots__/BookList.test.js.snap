// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book List should render correctly 1`] = `ShallowWrapper {}`;
